src/lexer.lex

```diff
@@ -62,63 +62,61 @@ COMMENT		(\$\$([^(\$\$)]*(\n?))*\$\$)|(\$[^(\$|\n)]*\$)
-
-{ASSIGN}      		{ fprintf(yyout, "ASSIGN\n"); lineCharacterCount += yyleng; }
-{ADD}			{ fprintf(yyout, "ADD\n"); lineCharacterCount += yyleng; }
-{SUB}			{ fprintf(yyout, "SUB\n"); lineCharacterCount += yyleng; }
-{DIV}			{ fprintf(yyout, "DIV\n"); lineCharacterCount += yyleng;}
-{MUL}			{ fprintf(yyout, "MUL\n"); lineCharacterCount += yyleng;}
-{MOD}			{ fprintf(yyout, "MOD\n"); lineCharacterCount += yyleng;}
-{EQ}			{ fprintf(yyout, "EQ\n"); lineCharacterCount += yyleng; }
-{LT}          		{ fprintf(yyout, "LT\n"); lineCharacterCount += yyleng; }
-{LTE}         		{ fprintf(yyout, "LTE\n"); lineCharacterCount += yyleng; }
-{GT}          		{ fprintf(yyout, "GT\n"); lineCharacterCount += yyleng; }
-{GTE}     		{ fprintf(yyout, "GTE\n"); lineCharacterCount += yyleng; }
-{NE}          		{ fprintf(yyout, "NE\n"); lineCharacterCount += yyleng; }
-
-{FUNCTION}		{ fprintf(yyout, "FUNCTION\n"); lineCharacterCount += yyleng;}
-{BEGINSCOPE}		{ fprintf(yyout, "BEGINSCOPE\n"); lineCharacterCount += yyleng;}
-{ENDSCOPE}		{ fprintf(yyout, "ENDSCOPE\n"); lineCharacterCount += yyleng;}
-{BEGINPARAM}		{ fprintf(yyout, "BEGINPARAM\n"); lineCharacterCount += yyleng;}
-{ENDPARAM}		{ fprintf(yyout, "ENDPARAM\n"); lineCharacterCount += yyleng;}
-{BEGINBRACKET}		{ fprintf(yyout, "BEGINBRACKET\n"); lineCharacterCount += yyleng;}
-{ENDBRACKET}		{ fprintf(yyout, "ENDBRACKET\n"); lineCharacterCount += yyleng;}
-{IF}          		{ fprintf(yyout, "IF\n"); lineCharacterCount += yyleng; }
-{ELSE}        		{ fprintf(yyout, "ELSE\n"); lineCharacterCount += yyleng; }
-{FOR}         		{ fprintf(yyout, "FOR\n"); lineCharacterCount += yyleng; }
-{WHILE}       		{ fprintf(yyout, "WHILE\n"); lineCharacterCount += yyleng; }
-{CONTINUE}    		{ fprintf(yyout, "CONTINUE\n"); lineCharacterCount += yyleng; }
-{BREAK}       		{ fprintf(yyout, "BREAK\n"); lineCharacterCount += yyleng; }
-{TRUE}			{ fprintf(yyout, "TRUE\n"); lineCharacterCount += yyleng;}
-{FALSE}			{ fprintf(yyout, "FALSE\n"); lineCharacterCount += yyleng;}
-{RETURN}		{ fprintf(yyout, "RETURN\n"); lineCharacterCount += yyleng;}
-{VOID}			{ fprintf(yyout, "VOID\n"); lineCharacterCount += yyleng;}
-{NOT}			{ fprintf(yyout, "NOT\n"); lineCharacterCount += yyleng;}
-{AND}			{ fprintf(yyout, "AND\n"); lineCharacterCount += yyleng;}
-{OR}			{ fprintf(yyout, "OR\n"); lineCharacterCount += yyleng;}
-{SEMICOLON}		{ fprintf(yyout, "SEMICOLON\n"); lineCharacterCount += yyleng; }
-{COMMA}			{ fprintf(yyout, "COMMA\n"); lineCharacterCount += yyleng;}
-
-{OUTPUT}		{ fprintf(yyout, "OUTPUT\n"); lineCharacterCount += yyleng;}
-{INPUT}			{ fprintf(yyout, "INPUT\n"); lineCharacterCount += yyleng;}
-
-{IDENTIFIER}   		{ fprintf(yyout, "IDENTIFIER: %s\n", yytext); lineCharacterCount += yyleng; }
-{BAD_IDENTIFIER}	{ printf("Error: Invalid identifier %s on line %d, col %d\n", yytext, lineCount + 1, lineCharacterCount + 1); lineCharacterCount += yyleng;}
+{INTEGER}       	{ return INTEGER }
```

Comment:
fprintf(yyout, "INTEGER \n"); lineCharacterCount += yyleng; should be here. If not our lineCharacterCount will be off whenever ints are declared.